### PR TITLE
Bug 1834956: Remove topology text filter from redux and rely on URL search params

### PR DESCRIPTION
**Pull request by rohitkrai03**
Fixes - https://issues.redhat.com/browse/ODC-3825?filter=12346946

Problem - Topology text filter is saved in redux which is why the filter is applied even if user returns to topology after being redirected from an action on another page.

Solution - Remove text filter from redux state completely and move the implementation based on URL search params only.

Screencast - 

Peek 2020-05-12 23-57

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx [hunk 1]
```diff
@@ -99,11 +121,12 @@ const mergeProps = (
     onFiltersChange({ ...filters, display });
   },
   onSearchQueryChange: (searchQuery) => {
-    onFiltersChange({ ...filters, searchQuery });
     if (searchQuery.length > 0) {
       setQueryArgument(TOPOLOGY_SEARCH_FILTER_KEY, searchQuery);
+      document.body.classList.add('odc-m-filter-active');
     } else {
       removeQueryArgument(TOPOLOGY_SEARCH_FILTER_KEY);
+      document.body.classList.remove('odc-m-filter-active');
```
Comment (jeff-phillips-18) [inline frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx:129]: I'd prefer to see this done in a callback for `onSeachChange` in the `Topology` component. It should probably be done _only_ there and removed from the `GraphComponent`.  That way it is controlled by a single component.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx:129]: You mean other class changes that are there in `GraphComponent` right now should also be moved to `Topology` component?

Comment (jeff-phillips-18) [reply frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx:129]: yes, as well as from a callback done here.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx]: @jeff-phillips-18 Done. Moved the logic into `onSearchChange` in `Topology`.

**Comment by andrewballantyne** [comment]
Hah, because of changes in `frontend/packages/kubevirt-plugin/src/topology/__tests__/kubevirt-topology-model.spec.ts`... this needs @christianvogt 

/assign @christianvogt

**Pull request merged**